src/eos-crash-metrics.c

```diff
@@ -137,6 +180,118 @@ normalize_path (char *path)
     path[i] = path[i] == '!' ? '/' : path[i];
 }
 
+static inline void
+sanitize_name (gchar *name)
+{
+  gchar *tmp = strchr (name, '/');
+
+  if (tmp != NULL)
+    *tmp = '\0';
+}
+
+static const char *
+get_associated_runtime (FlatpakInstalledRef *ref)
+{
+  g_autoptr(GBytes) metadata = NULL;
+  gchar *runtime = NULL;
+  g_autoptr(GKeyFile) key_file = NULL;
+  const gchar *metadata_data;
+  gsize metadata_len;
+  g_autoptr(GError) local_error = NULL;
+
+  metadata = flatpak_installed_ref_load_metadata (FLATPAK_INSTALLED_REF (ref), NULL, &local_error);
+
+  if (metadata == NULL)
+    {
+      g_warning ("%s", local_error->message);
+      return NULL;
+    }
+
+  metadata_data = g_bytes_get_data (metadata, &metadata_len);
+  key_file = g_key_file_new ();
+
+  if (!g_key_file_load_from_data (key_file, metadata_data, metadata_len, G_KEY_FILE_NONE, &local_error))
+    {
+      if (!g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        g_warning ("%s", local_error->message);
```

Comment:
Instead of doing this, you could probably bubble up the error using `g_error_propagate` and then report on failure in the caller if we failed to get the runtime for some reason.